feat: add customSlotComponent to ItemBuilder

ItsEmpa opened the pull request:
and add Component.toStringWithFormattingCodes function

currently the function doesn't actually optimize color codes, idk if thats something we should look at right now or do later

ItsEmpa committed:
feat: add customSlotComponent to ItemBuilder, and add Component.toStringWithFormattingCodes function

ThatGravyBoat commented on src/common/main/kotlin/tech/thatgravyboat/skyblockapi/utils/text/Text.kt:
```diff
@@ -153,6 +154,39 @@ object TextUtils {
     fun Component.substring(range: IntRange): Component = this.substring(range.first, range.last)
 
+    // TODO: optimize color codes to only add the necessary ones
+    fun Component.toStringWithFormattingCodes(): String {
```
make it internal, I dont want anyone using formatting codes in modern. If there is away you can just make it so vanilla can take a component for it thatd be better

ItsEmpa committed:
fix(VisualItem): fix customSlotText/Component not working

also makes visual items use components for custom slot text instead of formatted strings, and changes the toStringWithFormattingCodes function to be internal

ThatGravyBoat commented on src/versions/1.21.5/main/java/tech/thatgravyboat/skyblockapi/mixins/GuiGraphicsVisualItemMixin.java:
```diff
@@ -1,22 +1,26 @@
 import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
 import tech.thatgravyboat.skyblockapi.api.item.VisualItemAccessor;
 
+@Debug(export = true)
```
Please remove

ItsEmpa committed:
Merge branch '2.0' into feat/item-builder-slot-component

ItsEmpa committed:
fix: remove @Debug mixin

j10a1n15 commented on src/versions/1.21.5/main/java/tech/thatgravyboat/skyblockapi/mixins/GuiGraphicsVisualItemMixin.java:
```diff
@@ -42,29 +48,59 @@ public abstract class GuiGraphicsVisualItemMixin {
+            componentRef.set(slotText);
+        }
         else {
+            if (log) System.out.println("Diamond axe doesnt have slot text!");
```
i dont think this is intended to stay

ItsEmpa replied:
uhm oops :3

ItsEmpa committed:
fix: remove left over debugging
